pandas/plotting/_matplotlib/core.py

```diff
@@ -767,8 +767,10 @@ def _get_index_name(self) -> str | None:
                 name = pprint_thing(name)
 
         # GH 9093, override the default xlabel if xlabel is provided.
-        if self.xlabel is not None:
+        if self.xlabel is not None and self.orientation == "vertical":
             name = pprint_thing(self.xlabel)
+        if self.ylabel is not None and self.orientation == "horizontal":
```

Comment:
so if these are provided and there is an incorrect orientation it is ignored? maybe could add a warning for that case

Reply:
Done &ndash; added a warning to both cases.

Reply:
Actually, I don't think it makes sense to have the warning here. Before this PR, running the below would have caused the `xlabel` argument to be used for the y-axis:

```python
import pandas as pd

df = pd.DataFrame(
    {'A': [4, 2, 1, 2, 5, 2]}, index=pd.RangeIndex(6, name='B')
)
df.plot(kind='barh', xlabel='x')
```

A warning for this case with the new code doesn't seem correct because now by "ignoring" the `xlabel`/`ylabel` values at this point in the code, we are just ensuring that when the axis labels are applied later it is done correctly. For example, with the new code we get a warning, but the output is exactly what was intended:
<img width="767" alt="Screen Shot 2022-01-10 at 6 02 31 PM" src="https://user-images.githubusercontent.com/24376333/148852527-310d54df-8d9f-485f-86df-c4a432e873e1.png">